saml2/saml2-service-provider/src/main/java/org/springframework/security/saml2/provider/service/web/Saml2AuthenticationTokenConverter.java

```diff
@@ -110,6 +112,7 @@ private String inflateIfRequired(HttpServletRequest request, byte[] b) {
 
 	private byte[] samlDecode(String base64EncodedPayload) {
 		try {
+			BASE_64_CHECKER.checkAcceptable(base64EncodedPayload);
```

Comment:
What purpose does `BASE_64_CHECKER` serve? I imagine that if `BASE64.decode` fails, then the exception handling will be the same as before.

Reply:
The old decoder was more strict about rejecting possibly invalid Base64 strings. The checker will reject those cases even though `java.util.Base64` would accept it. There is a test that fails if this checking is not present.

Reply:
The other option would be to remove the failing test, but I can't assess the security significance of that change.

Reply:
Sounds good. We can always relax the check later on if needed.